test/python/compiler/test_assembler.py

```diff
@@ -868,5 +871,36 @@ def test_single_and_deprecated_acquire_styles(self):
         self.assertEqual(assembled_acquire.memory_slot, [0, 1, 2, 3, 4])
 
 
+class StreamHandlerRaiseException(StreamHandler):
+    """Handler class that will raise an exception on formatting errors."""
+
+    def handleError(self, record):
+        raise sys.exc_info()
+
+
+class TestLogAssembler(QiskitTestCase):
+    """Testing the log_assembly option."""
+
+    def setUp(self):
+        logger = getLogger()
+        logger.setLevel('DEBUG')
+        self.output = io.StringIO()
+        logger.addHandler(StreamHandlerRaiseException(self.output))
+        self.circuit = QuantumCircuit(QuantumRegister(1))
+
+    def assertAssembleLog(self, log_msg):
+        """ Runs assemble and checks for logs containing specified message"""
+        assemble(self.circuit, shots=2000, memory=True)
+        self.output.seek(0)
+        # Filter unrelated log lines
+        output_lines = self.output.readlines()
+        assembly_log_lines = [x for x in output_lines if x.__contains__(log_msg)]
```

Comment:
Is there a reason not to do:

```suggestion
        transpile_log_lines = [x for x in output_lines if log_msg in x]
```
Calling the private methods seems a bit weird.